internal/trie/node/dirty.go

```diff
@@ -5,20 +5,20 @@ package node
 
 // IsDirty returns the dirty status of the branch.
 func (b *Branch) IsDirty() bool {
-	return b.dirty
+	return b.Dirty
 }
 
 // SetDirty sets the dirty status to the branch.
 func (b *Branch) SetDirty(dirty bool) {
```

Comment:
should we keep maintaining this method once `Dirty` is an exported value?

Reply:
Yes if you want to check on the node interface without doing type assertions on branch/leaf.

Also note I exported it to write tests in `lib/trie` otherwise it's really a pain to write test code. This is not really for production code purposes.